ui/app/routes/allocations/allocation/task/fs.js

```diff
@@ -0,0 +1,33 @@
+import Route from '@ember/routing/route';
+import RSVP from 'rsvp';
+
+export default Route.extend({
+  model({ path = '/' }) {
+    const decodedPath = decodeURIComponent(path);
+    const task = this.modelFor('allocations.allocation.task');
+
+    const pathWithTaskName = `${task.name}${decodedPath.startsWith('/') ? '' : '/'}${decodedPath}`;
+
+    return task.stat(pathWithTaskName).then(statJson => {
+      if (statJson.IsDir) {
+        return RSVP.hash({
+          path: decodedPath,
+          task,
+          directoryEntries: task.ls(pathWithTaskName),
+          isFile: false,
+        });
+      } else {
+        return {
+          path: decodedPath,
+          task,
+          isFile: true,
```

Comment:
Do you think it's worth adding a `directoryEntries: []` here just to ensure that the controller prop is always an array?

I'm on the fence about it.

Reply:
hmm… I don’t really even like this giant-conditional-in-the-template situation but I feel like `isFile` adequately protects against any attempt to iterate through `directoryEntries`. I’d be fine adding it though if you think it’s helpful!

Reply:
Nah, no need.